### PR TITLE
Simplify virtualenv testing

### PR DESCRIPTION
**Pull request by kubouch**
# Description

Instead of running virtualenv's test suite, run only Nushell's test script (not including pydoc test, not sure how to get it working without the virtualenv's test suite).

# User-Facing Changes

# Tests + Formatting

# After Submitting

### CODE REVIEW

File: scripts/test_virtualenv.nu [hunk 1]
```diff
@@ -0,0 +1,43 @@
+let env_name = 'e-$ èрт🚒♞中片-j'
```
Comment (hyiltiz) [inline scripts/test_virtualenv.nu:1]: Scrambled characters as `env_name`, intended?

Comment (kubouch) [reply scripts/test_virtualenv.nu:1]: Yes, it's taken from virtualenv test suite

**Comment by amtoine** [comment]
@kubouch 
how close do you think we are with this PR? :innocent:

**Comment by kubouch** [comment]
It's done, we just need to figure out why Windows is failing.

**Comment by amtoine** [comment]
> It's done
>
ooh that's great :partying_face: 

> we just need to figure out why Windows is failing.
>
mm ok i see it now :thinking:

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -140,22 +140,16 @@ jobs:
...
       - name: Test Nushell in virtualenv
-        run: |
-          cd virtualenv
-          # if we encounter problems with bleeding edge tests pin to the latest tag
-          # git checkout $(git describe --tags | cut -d - -f 1)
-          # We need to disable failing on coverage levels.
-          nu -c "open pyproject.toml | upsert tool.coverage.report.fail_under 1 | save patchproject.toml"
-          mv patchproject.toml pyproject.toml
-          tox -e ${{ matrix.py }} -- -k nushell
+        run: cd scripts; nu test_virtualenv.nu
```
Comment (amtoine) [inline .github/workflows/ci.yml:152]: can we just do
```suggestion
        run: nu scripts/test_virtualenv.nu
```
here?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -140,22 +140,16 @@ jobs:
         with:
           python-version: "3.10"
 
-      - run: python -m pip install tox
+      #- run: python -m pip install tox
```
Comment (amtoine) [inline .github/workflows/ci.yml:143]: maybe we can get rid of this comment? :innocent:

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -140,22 +140,16 @@ jobs:
...
 
         # Get only the latest tagged version for stability reasons
       - name: Install virtualenv
-        run: git clone https://github.com/pypa/virtualenv.git
+        #run: git clone https://github.com/pypa/virtualenv.git
```
Comment (amtoine) [inline .github/workflows/ci.yml:147]: and this comment too? :innocent:

**Review by amtoine — CHANGES_REQUESTED** [review]

**Comment by kubouch** [comment]
So now it doesn't work at all. Not sure why.

File: scripts/test_virtualenv.nu [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+let test_lines = [
+    "python -c 'import sys; print(sys.executable)'"
+    "python -c 'import os; import sys; v = os.environ.get("VIRTUAL_ENV"); print(v)'"
+    $"overlay use '([$env.PWD $env_name bin activate.nu] | path join)'"
+    "python -c 'import sys; print(sys.executable)'"
+    "python -c 'import os; import sys; v = os.environ.get("VIRTUAL_ENV"); print(v)'"
+    "print $env.VIRTUAL_PROMPT"
```
Comment (amtoine) [inline scripts/test_virtualenv.nu:9]: the CI says this should be `VIRTUAL_ENV_PROMPT` :thinking:

**Comment by amtoine** [comment]
do we give up on this @kubouch ? :cry:

**Comment by amtoine** [comment]
> Oh no, I thought I deleted some random fix branch! Should not be deleted 🤣
>
:scream: 

feel free to restore the branch and reopen the PR then before forgetting about it :wink:

**Pull request closed without merging**